Source/WebCore/editing/TextIterator.cpp

```diff
@@ -743,28 +744,28 @@ bool TextIterator::handleReplacedElement()
     if (m_fullyClippedStack.top())
         return false;
 
-    auto& renderer = *m_node->renderer();
+    auto& renderer = *m_currentNode->renderer();
     if (renderer.style().visibility() != Visibility::Visible && !m_behaviors.contains(TextIteratorBehavior::IgnoresStyleVisibility))
         return false;
 
     if (m_lastTextNodeEndedWithCollapsedSpace) {
-        emitCharacter(' ', *m_lastTextNode->parentNode(), m_lastTextNode, 1, 1);
+        emitCharacter(' ', Ref { *m_lastTextNode->parentNode() }, RefPtr { m_lastTextNode.get() }.get(), 1, 1);
         return false;
     }
 
     if (m_behaviors.contains(TextIteratorBehavior::EntersTextControls) && is<RenderTextControl>(renderer)) {
         if (auto innerTextElement = downcast<RenderTextControl>(renderer).textFormControlElement().innerTextElement()) {
-            m_node = innerTextElement->containingShadowRoot();
-            pushFullyClippedState(m_fullyClippedStack, *m_node);
+            m_currentNode = innerTextElement->containingShadowRoot();
+            pushFullyClippedState(m_fullyClippedStack, *protectedCurrentNode());
             m_offset = 0;
             return false;
         }
     }
 
-    if (m_behaviors.contains(TextIteratorBehavior::EntersImageOverlays) && is<HTMLElement>(m_node) && ImageOverlay::hasOverlay(downcast<HTMLElement>(*m_node))) {
-        if (RefPtr shadowRoot = m_node->shadowRoot()) {
-            m_node = shadowRoot.get();
-            pushFullyClippedState(m_fullyClippedStack, *m_node);
+    if (m_behaviors.contains(TextIteratorBehavior::EntersImageOverlays) && is<HTMLElement>(m_currentNode.get()) && ImageOverlay::hasOverlay(downcast<HTMLElement>(*m_currentNode))) {
+        if (RefPtr shadowRoot = m_currentNode->shadowRoot()) {
+            m_currentNode = shadowRoot.get();
```

Comment:
Why is `get()` needed here?

Reply:
You can't assign a RefPtr to a WeakPtr currently. You have to get a raw pointer from the RefPtr first.